### PR TITLE
Provider Pictogromas, Rutas con Argumentos y mejora de diseño

### PR DESCRIPTION
**Pull request by LordMario**
## Description
* Implementacion de provider para la lista de pictogramas
* Refactorizacion de inputs y selectBox para uso de maneja generica
* Agregando rutas para la pantalla de child y lista de pictogramas con argumento(Data)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (no functional changes, no api changes)
- [ ] This change requires a documentation update
- [ ] Other (please describe): 

## How Has This Been Tested?

Describe the tests you've run to verify the changes. Provide instructions for reproduction and indicate any relevant details for your test setup.

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in areas hard to understand.
- [ ] I have made the corresponding changes to the documentation.
- [x] My changes do not generate new warnings.
- [ ] I have added tests that prove my solution is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] All dependent changes have been merged and published in subsequent modules.
- [x] I have reviewed my code and corrected any spelling mistakes.

### CODE REVIEW

File: lib/presentation/pages/children/childs_data.page.dart
Comment (SamBarFu) [inline lib/presentation/pages/children/childs_data.page.dart:1]: Cambiar el nombre del archivo

File: lib/presentation/pages/pictogram/pictogram.page.dart [hunk 1]
```diff
@@ -34,10 +35,11 @@ class PictogramPage extends StatelessWidget {
...
-        title: Text(S.current.Pictogramas_personalizados),
+        title: const Text(
+            'Lista de pictogramas generales'), //TODO : Cambiar con la internacionalizacion en el siguiente commit
       ),
       body: GridImages(
-        isCustomized: true,
+        isCustomized: false,
         images: arregloImages,
         // ignore: avoid_print
         loadNextImages: () => print('cargando nuevas imagenes'),
```
Comment (SamBarFu) [inline lib/presentation/pages/pictogram/pictogram.page.dart:45]: agregar internacionalización

**Review by SamBarFu — CHANGES_REQUESTED** [review]

File: lib/presentation/pages/pictogram/pictogram.page.dart (hunk 1 above)
Comment (LordMario) [reply lib/presentation/pages/pictogram/pictogram.page.dart:45]: Eso es una función de prueba, arriba hay un comentario que se debe de quitar cuando este listo el endpoint, no se ve el comentario porque no es un cambio en el commit

File: lib/presentation/pages/children/childs_data.page.dart
Comment (LordMario) [reply lib/presentation/pages/children/childs_data.page.dart:1]: Listo

**Review by SamBarFu — APPROVED** [review]

**Pull request merged**